### ISSUE TITLE
Would you like to become a patron of language-ext?

### ISSUE THREAD

**Issue by louthy** [louthy/language-ext#223]
I'm Paul Louth and the author of language-ext. My goal for this project is to create a BCL for functional programming in C#. I want to take this as far as it can go, and also be useful as leverage when pushing for new functional features in C#. 

_As you can probably imagine it takes an inordinate amount of my spare time!_

Maintaining and developing the features of this project is a labour of love without a doubt, and I will continue to build this for personal reasons no matter what. However, if you run a business and are using language-ext in a revenue-generating product, it would make business sense to sponsor language-ext development: it ensures the project that your product relies on stays healthy and actively maintained. 

Of course, individual users are also welcome to make a recurring pledge if language-ext has helped you in your work or personal projects.

https://www.patreon.com/louthy

**Comment by Maximusya** [comment]
Any option for a one-time payment? Maybe it would not be too helpful - just a way to make virtual beer a real beer)

**Comment by louthy** [comment]
@Maximusya Yeah sure, I've set up a [ko-fi page](http://ko-fi.com/louthy), for virtual coffee - I'm more of a coffee addict than a beer addict ;)

Thanks for the support!

**Comment by longqinsi** [comment]
I have support you 3$. Sorry because of my incoming, I can't give you too much.

**Comment by longqinsi** [comment]
Maybe I will give you more sponsor, but a regular patreon like www.patreon.com is too heavy for me. I can't give a promise to give how much money at some time. But maybe sometime later when I come here again I will sponsor another virtual coffee.
Anyhow, the language-ext you created and are maintaining is great. Keep on.

**Comment by longqinsi** [comment]
BTW, there have been several books about functional programming in c# in the market. I think you can contact them to see if anyone of them wants to write a book for your framework.

**Comment by longqinsi** [comment]
Or you can contact some IT publisher, e.g. O'Reilly, Packt Publishing, ane etc.

**Comment by longqinsi** [comment]
BTW, instead of add more function to language-ext, I think you can take a while to write some samples for it. I have seen that the samples project in the solution is empty. If there are some pragmatic samples, I think there will be more users of your library, and maybe more patrons.
Another thing, you can consider become an author in pluralsight.com by teaching how to use language-ext. It will take you a large amount of time, but it will also lower the threhold for people to master language-ext. And the lessons in pluralsight are for no extra fee for subsribers of pluralsight, unlike udemy.com where every lesson has its own price. This means that if you publish a lesson on pluragsight, every subsriber can immediately access it without any obstacle. And then the lesson can become an advertisement of language.
Of course, if you like, you can publish lesson to udemy.com.
Maybe you can publish some introduction or tutorial video of language-ext on Youtube, Channel9, and so on.
In general, I think the most thing for language-ext at present, is not to make it better, but to make it known to more people and easier to use by others.

**Comment by louthy** [comment]
> Or you can contact some IT publisher, e.g. O'Reilly, Packt Publishing, and etc.

I've already been asked to write a book on functional programming.  I just don't have the free time to be a CTO, develop lang-ext, and write a book (as well as the non-coding related interests I have in my life).  If there were 48 hours in a day I might consider it ;)

>  think you can take a while to write some samples for it. I have seen that the samples project in the solution is empty. If there are some pragmatic samples, I think there will be more users of your library, and maybe more patrons.

Whilst I agree that more samples is a good idea, it is often difficult to demonstrate the scope of a project like this.  Mostly because it is all very abstract.  It's a bit like writing samples on how to use `int`.  This isn't a repo for teaching functional programming.  It's a repo to provide the common tools of functional programming that are found in other functional languages.  I realise that a lot of C# devs haven't touched functional programming before, or are inexperienced, and in those circumstances I'm always happy to help through the issues section.  But also I'm not here to explain this stuff.  I provide the toolkit, it's up to you to go and learn functional programming.  Apart from the type-class stuff there's nothing really 'surprising' here in terms of functionality, and the type-class stuff can be pretty much ignored unless you're at an advanced level.

I think something like the fsharpforfunandprofit tutorials would be quite good in this situation, but it would require a lot of time.  I am always open to people contributions on this front (that can added to the wiki).  So if anyone is reading this and wants to get involved in adapting fsharpforfunandprofit-like tutorials, I would be very grateful.

**Comment by longqinsi** [comment]
Ok, I'm sorry that I didn't think of your circumstances as a CTO who have been very busy.
I'm still studying functional programming and your framework. I will keep a key on your language-ext. It's great!

**Comment by cjdsellers** [comment]
Hi Louthy,

I'm currently writing a trading platform I hope to go open source with. I've been finding good use for more and more functional approaches within the code base, I was just saying to a mate a few days ago that it would be great if there was even more support for functional concepts in C#. I was starting to do some some more research and then stumbled upon your library! I'm extremely impressed in its scope, and honestly it's probably far more than what I need (although I appreciate the wide vision you have for language-ext). If I find any financial success through my use of, or inspiration from language-ext, I'll be sure to pay credit where credit (and pay) are indeed due.

Cheers,
Chris

**Comment by louthy** [comment]
@cjdsellers Hi Chris, thanks for the kind words.  Good luck with your project :)

**Comment by abdulkaderjeelani** [comment]
Hi Louthy,
Hats off for writing a wonderful library, Its just great!
It would be great if you could provide some pointers on using interpreters pattern, for i/o operation, 
best approach to do database operations.
Thanks
